CMakeLists.txt

```diff
@@ -158,17 +159,51 @@ endif()
 
 # Targets ###################
 
+include(GNUInstallDirs)
+
 find_package(Threads REQUIRED)
-add_library(rlbox-cpp17-lib INTERFACE)
-target_include_directories(rlbox-cpp17-lib INTERFACE code/include)
-target_link_libraries(rlbox-cpp17-lib INTERFACE ${CMAKE_THREAD_LIBS_INIT})
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME} INTERFACE
+  $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/code/include>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_17)
+target_link_libraries(${PROJECT_NAME} INTERFACE ${CMAKE_THREAD_LIBS_INIT})
+
+# Install ###################
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file("${PROJECT_NAME}ConfigVersion.cmake"
+                                 VERSION ${CMAKE_PROJECT_VERSION}
+                                 COMPATIBILITY SameMajorVersion)
+
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${RLBOX_SOURCE_FILES}")
+
+set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_INCLUDEDIR}")
+set(DATAROOT_CONFIG_DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/cmake/${PROJECT_NAME}")
+configure_package_config_file(${PROJECT_SOURCE_DIR}/cmake/${PROJECT_NAME}Config.cmake.in
+  ${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
+  INSTALL_DESTINATION ${DATAROOT_CONFIG_DESTINATION}
+  PATH_VARS INCLUDE_INSTALL_DIR)
+install(TARGETS ${PROJECT_NAME}
```

Comment:
The catch2 projects seem to run install only when they aren't included as a subdirectory due to bug https://github.com/catchorg/Catch2/issues/1373

See https://github.com/catchorg/Catch2/blob/devel/CMakeLists.txt#L109
We should probably do the same here

Reply:
I looked at the bug. I uninstall system catch2 to force local catch2.

After a:

```
make DESTDIR=/tmp/ttt install
cd /tmp/ttt/
find
```

I have:

```
.
./usr
./usr/local
./usr/local/share
./usr/local/share/cmake
./usr/local/share/cmake/rlbox-sandboxing-api
./usr/local/share/cmake/rlbox-sandboxing-api/rlbox-sandboxing-apiTargets.cmake
./usr/local/share/cmake/rlbox-sandboxing-api/rlbox-sandboxing-apiConfig.cmake
./usr/local/share/cmake/rlbox-sandboxing-api/rlbox-sandboxing-apiConfigVersion.cmake
./usr/local/rlbox
./usr/local/rlbox/rlbox_range.hpp
./usr/local/rlbox/rlbox_type_traits.hpp
./usr/local/rlbox/rlbox_sandbox.hpp
./usr/local/rlbox/rlbox_stdlib_polyfill.hpp
./usr/local/rlbox/rlbox_struct_support.hpp
./usr/local/rlbox/rlbox_stdlib.hpp
./usr/local/rlbox/rlbox_helpers.hpp
./usr/local/rlbox/rlbox_types.hpp
./usr/local/rlbox/rlbox_unwrap.hpp
./usr/local/rlbox/rlbox_noop_sandbox.hpp
./usr/local/rlbox/rlbox_wrapper_traits.hpp
./usr/local/rlbox/rlbox_policy_types.hpp
./usr/local/rlbox/rlbox.hpp
./usr/local/rlbox/rlbox_app_pointer.hpp
./usr/local/rlbox/rlbox_conversion.hpp
./usr/local/rlbox/rlbox_dylib_sandbox.hpp
```

So I do not have files from catch2.

According to the ticket from catch2, it appends only if `include(GNUInstallDirs)` is previous `add_subdirectory("${catch2_SOURCE_DIR}")`. It's not the case if rlbox.

I also tried to move `include(GNUInstallDirs)` before `find_package(Catch2 QUIET)` and I don't have files from catch2 neither.

Did you had this bug ?

Reply:
Oh i meant a similar thing would happen if someone included the rlbox project as a sub_project and then ran make install in their project, then rlbox may get installed in a sub directory there (like catch does). Its not a big deal either way though --- i think the fix is to disable the install target if this is a sub_project as shown in the catch repo link